src/main/java/org/testeditor/web/xtext/index/persistence/GitService.xtend

```diff
@@ -38,4 +61,13 @@ class GitService {
 		}
 	}
 
+	private def void cloneRepository(File projectFolder, String uriString) {
+		git = Git.cloneRepository.setDirectory(projectFolder).setURI(uriString).call
+	}
+
+	@VisibleForTesting
```

Comment:
openRepository is a side-effect of calling init (with the right preconditions). I assume you want to avoid a call to `pull`? Wouldn't it be possible / cleaner to mock the dependency on Git...?

Reply:
touched a nerve here. I will have to look into this, again